cypress/support/TestListManager.js

```diff
@@ -3,225 +3,143 @@
-      { name: "Book me a Flight", func: train.BookMeAFlight}
-    ]
-  },
-  {
-    name: 'CleanUp', tests:
-    [
-      { name: 'Delete All Test Generated Models', func: deleteAllTestGeneratedModels.DeleteAllTestGeneratedModels},
-    ]
-  },
+// This is accomplished by doing the following:
+//  1) Place all .js files, which should contain multiple test cases, in 
+//     the "cypress/tests" folder.
+//  2) For each test case, directly before the test function, call the
+//     following function with appropriate parameters:
+//        Cypress.TestCase(testGroupName, testDescription, testFunction)
+//  3) From the command line run: 
+//        node cypress\GenerateCypressTestSpecs.js
+//     (It is intended that this be automated, but at the moment it is not.)
+// ----------------------------------------------------------------------
+
+var testLists = require('../TestLists');
+const helpers = require('./Helpers');
+
+Cypress.TestCase = TestCase;
+Cypress.testList = testLists.testList;
+Cypress.regressionTestList = testLists.regressionTestList;
+Cypress.masterListOfAllTestCases = testLists.masterListOfAllTestCases;
+
+var testGroups = new Array();
+
+// ----------------------------------------------------------------------
+// NOTE: Placement of these "require" statements is important in this file.
+//
+// Each of these required files will call back into the TestCase function
+// found in this file.
+// ----------------------------------------------------------------------
+
+const createModels = require('../tests/CreateModels');
+const editAndBranching = require('../tests/EditAndBranching');
+const log = require('../tests/Log');
+const train = require('../tests/Train');
+const tools = require('../tests/Tools');
+const zTemp = require('../tests/zTemp')
+
+// ----------------------------------------------------------------------
+// Call this function to add an ordinary function to the list of test 
+// cases that Cypress will recognize as a test. This technique is needed
+// to support running multiple tests from any spec file in any order.
+// ----------------------------------------------------------------------
+
+function TestCase(testGroupName, testDescription, testFunction) {
+  // These two lines of code converts the testFunction from a callable
+  // function to a string containing the name of this function. It works
+  // like this:
+  //   1) We convert the function to a string containing the full function
+  //      that looks like: 
+  //        "function TheTestFunctionToRun(param1, param2) {...}"
+  //   2) The magic number 9 here is skipping past the string "function "
+  //      and grabbing just the name of the function.
+  var testFunctionAsString = testFunction.toString();
+  var testFunctionName = testFunctionAsString.substring(9, testFunctionAsString.indexOf('(', 9));
+  helpers.ConLog('TestListManager', `TestCase(${testGroupName}, ${testDescription}, ${testFunctionName})`);
```

Comment:
Similar to the comments I made about the `Cypress.TestCase` searching. Would at least extract the magic number to a variable and look at using RegEx.

E.g.
```
const functionPrefix = `function `
const testFunctionName = testFunctionAsString.substring(functionPrefix.length, testFunctionAsString.indexOf('(', functionPrefix.length))
```

It's better to code self document as much as possible. The comments could get stale, and we're optimizing for readability over performance.

Reply:
Regex is overkill here, and not very easy to understand. Most people I know have to look up Regex strings to understand them.

I do agree with your comment about comments going stale, which is why I so rarely use them, but in this case I feel that this code benefits from the comment. I did however, fix the code to use the length as you suggested, that works for me.